src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        // Формат дат
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter);
+
+        Map<String, Integer> salaryMap = new HashMap<>();
+        for (String name : names) {
+            salaryMap.put(name, 0);
+        }
+
+        for (String entry : data) {
```

Comment:
Consider adding a check to ensure that each entry in the `data` array has the expected format and number of parts before attempting to parse it. This will help avoid potential `ArrayIndexOutOfBoundsException` if the data is malformed.